File: files/arc_openmp/c/setthreads.c
// Load the OpenMP functions library
#include<omp.h>

int main()
{
  // Set variables
  int tnum=0;
    
  // Create a parallel block of four threads (including master thread)
/* #pragma omp parallel private(tnum) num_threads(4) */
/*   {     */
    tnum = omp_get_thread_num();
    printf("I am thread number: %d\n", tnum);
  /* } */

  // Create a parallel block, without specifying the number of threads
/* #pragma omp parallel private(tnum) */
/*   {     */
    tnum = omp_get_thread_num();
    printf("Second block, I am thread number %d\n", tnum);
  /* } */
    
  return 0;
}

File: files/arc_openmp/fortran/setthreads.f90
program set_threads

    ! Load the OpenMP functions library
    use omp_lib

    ! Set variables
    implicit none
    integer :: tnum

    ! Create a parallel block of four threads (including master thread)
    !$OMP PARALLEL PRIVATE(tnum) NUM_THREADS(4)
        tnum = OMP_GET_THREAD_NUM()
        print *, "I am thread number", tnum
    !$OMP END PARALLEL

    ! Create a parallel block, without specifying the number of threads
    !$OMP PARALLEL PRIVATE(tnum)
        tnum = OMP_GET_THREAD_NUM()
        print *, "Second block, I am thread number", tnum
    !$OMP END PARALLEL

end program set_threads
